### PR TITLE
Add `tiktoken` package

### LINKED ISSUE #3875: tiktoken need to be compiled into wasm
https://github.com/openai/tiktoken.
in fact,its file "lib.rs" is easy to be compiled to wasm.
so it can be supported?

### LINKED ISSUE #4245: Package Request - tiktoken v0.5.1
## 🐍 Package Request

- tiktoken 0.5.1
- https://github.com/openai/tiktoken/
- Package Dependencies that needs to be resolved first:
```
dependencies = ["regex>=2022.1.18", "requests>=2.26.0"]
optional-dependencies = {blobfile = ["blobfile>=2"]}
```

I've tried to make a wheel https://github.com/psymbio/tiktoken_rust_wasm/blob/main/packages/build_emscripten/tiktoken-0.5.1-cp311-cp311-emscripten_3_1_45_wasm32.whl and look at the [README](https://github.com/psymbio/tiktoken_rust_wasm/blob/main/README.md) for steps to create the wheel but I get this error: `ImportError: Can't connect to HTTPS URL because the SSL module is not available.` because the `requests` library needs to be figured out (https://github.com/pyodide/pyodide/issues/398), I'm open to hacky-patches but would like to know how to deal with it.

### PR DESCRIPTION
**Pull request by riyavsinha**
### Description

The `tiktoken` package by OpenAI is a popular tokenization library that would be great to have support for in Pyodide. Doing so will help enable LLM agent applications to be built and run in WASM.

Unfortunately, the tiktoken package maintainer is not interested in supporting Pyodide: https://github.com/openai/tiktoken/issues/94#issuecomment-1498263172

Resolve #4245
Resolve #3875

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

File: packages/tiktoken/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - rustup
+  run:
+    - regex
+    - requests
+about:
+  home: null
+  PyPI: https://pypi.org/project/tiktoken
+  summary: tiktoken is a fast BPE tokeniser for use with OpenAI's models
+  license:
+    'MIT License  Copyright (c) 2022 OpenAI, Shantanu Jain  Permission is hereby
```
Comment (ryanking13) [inline packages/tiktoken/meta.yaml:21]: Let's change it to simply

```
license: MIT
```

File: packages/tiktoken/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  url: https://files.pythonhosted.org/packages/37/02/576ff3a6639e755c4f70997b2d315f56d6d71e0d046f4fb64cb81a3fb099/tiktoken-0.8.0.tar.gz
+  sha256: 9ccbb2740f24542534369c5635cfd9b2b3c2490754a78ac8831d99f89f94eeb2
+requirements:
+  executable:
+    - rustup
+  run:
+    - regex
+    - requests
+about:
+  home: null
```
Comment (ryanking13) [inline packages/tiktoken/meta.yaml:17]: Let's use the same path for PyPI.

```
https://pypi.org/project/tiktoken
```

**Review by ryanking13 — APPROVED** [review]
Thanks! I have some minor comments, otherwise looks good to me.

File: packages/tiktoken/meta.yaml (hunk 2 above)
Comment (riyavsinha) [reply packages/tiktoken/meta.yaml]: Done!

File: packages/tiktoken/meta.yaml (hunk 1 above)
Comment (riyavsinha) [reply packages/tiktoken/meta.yaml]: Done :)

**Comment by hoodmane** [comment]
Thanks @riyavsinha!

**Pull request merged**